### PR TITLE
Add table aws_secrets_manager_secrets. closes #311

### LINKED ISSUE #311: Add table aws_secrets manager_secrets
**References**
Need to validate :- 
[SecretsManager.1] This control checks whether a secret stored in AWS Secrets Manager is configured to rotate automatically from `AWS Foundational Security Best Practices v1.0.0` 
SecretsManager.2:- This control checks whether an AWS Secrets Manager secret rotated successfully based on the rotation schedule. The control fails if RotationOccurringAsScheduled is false. The control does not evaluate secrets that do not have rotation configured

https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html#secretsmanager-1-remediation

image

### PR DESCRIPTION
**Pull request by sayan133**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_secrets_manager_secret []

PRETEST: tests/aws_secrets_manager_secret

TEST: tests/aws_secrets_manager_secret
Running terraform
aws_secretsmanager_secret.named_test_resource: Creating...
aws_secretsmanager_secret.named_test_resource: Creation complete after 6s [id=arn:aws:secretsmanager:us-east-1:388460667113:secret:turbottest93948-ZZ2Wmt]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Warning: Deprecated Attribute

  on variables.tf line 49, in resource "aws_secretsmanager_secret" "named_test_resource":
  49: resource "aws_secretsmanager_secret" "named_test_resource" {

Use the aws_secretsmanager_secret_rotation resource instead


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = "388460667113"
aws_partition = "aws"
aws_region = "us-east-1"
resource_aka = "arn:aws:secretsmanager:us-east-1:388460667113:secret:turbottest93948-ZZ2Wmt"
resource_name = "turbottest93948"

Running SQL query: test-get-query.sql
[
  {
    "account_id": "388460667113",
    "akas": [
      "arn:aws:secretsmanager:us-east-1:388460667113:secret:turbottest93948-ZZ2Wmt"
    ],
    "arn": "arn:aws:secretsmanager:us-east-1:388460667113:secret:turbottest93948-ZZ2Wmt",
    "deleted_date": null,
    "description": "Test secret.",
    "kms_key_id": null,
    "last_accessed_date": null,
    "last_rotated_date": null,
    "name": "turbottest93948",
    "owning_service": null,
    "partition": "aws",
    "primary_region": null,
    "region": "us-east-1",
    "replication_status": null,
    "rotation_enabled": null,
    "rotation_lambda_arn": null,
    "rotation_rules": null,
    "secret_versions_to_stages": null,
    "tags": {
      "foo": "bar"
    },
    "tags_src": [
      {
        "Key": "foo",
        "Value": "bar"
      }
    ],
    "title": "turbottest93948"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:secretsmanager:us-east-1:388460667113:secret:turbottest93948-ZZ2Wmt",
    "name": "turbottest93948",
    "region": "us-east-1"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "388460667113",
    "akas": [
      "arn:aws:secretsmanager:us-east-1:388460667113:secret:turbottest93948-ZZ2Wmt"
    ],
    "arn": "arn:aws:secretsmanager:us-east-1:388460667113:secret:turbottest93948-ZZ2Wmt",
    "name": "turbottest93948",
    "region": "us-east-1",
    "tags": {
      "foo": "bar"
    },
    "title": "turbottest93948"
  }
]
✔ PASSED

POSTTEST: tests/aws_secrets_manager_secret

TEARDOWN: tests/aws_secrets_manager_secret

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
### Basic info

```sql
select
  name,
  created_date,
  description,
  last_accessed_date
from
  aws_secrets_manager_secret;
```
```
+-----------+---------------------+-----------------+---------------------+
| name      | created_date        | description     | last_accessed_date  |
+-----------+---------------------+-----------------+---------------------+
| stmp      | 2021-04-20 14:07:58 | Test secret     | <null>              |
| test-stmp | 2021-04-20 06:26:24 | testing secret. | 2021-04-20 00:00:00 |
+-----------+---------------------+-----------------+---------------------+
```

### List secrets whose automatic rotation is not enabled

```sql
select
  name,
  created_date,
  description,
  rotation_enabled
from
  aws_secrets_manager_secret
where
  rotation_enabled = 'false';
```
```
+-----------+---------------------+-----------------+------------------+
| name      | created_date        | description     | rotation_enabled |
+-----------+---------------------+-----------------+------------------+
| test-stmp | 2021-04-20 06:26:24 | testing secret. | false            |
+-----------+---------------------+-----------------+------------------+
```

### List secrets whose automatic rotation interval is greater tha 7 days

```sql
select
  name,
  created_date,
  description,
  rotation_enabled,
  rotation_rules
from
  aws_secrets_manager_secret
where
  rotation_rules -> 'AutomaticallyAfterDays' > '7';
```
```
+-----------+---------------------+-----------------+------------------+-------------------------------+
| name      | created_date        | description     | rotation_enabled | rotation_rules                |
+-----------+---------------------+-----------------+------------------+-------------------------------+
| stmp      | 2021-04-20 14:07:58 | Test secret     | true             | {"AutomaticallyAfterDays":30} |
| test-stmp | 2021-04-20 06:26:24 | testing secret. | false            | {"AutomaticallyAfterDays":30} |
+-----------+---------------------+-----------------+------------------+-------------------------------+
```

### List secrets which are not replicated in other regions

```sql
select
  name,
  created_date,
  description,
  replication_status
from
  aws_secrets_manager_secret
where
  replication_status is NULL;
```
```
+------+---------------------+-------------+--------------------+
| name | created_date        | description | replication_status |
+------+---------------------+-------------+--------------------+
| stmp | 2021-04-20 14:07:58 | Test secret | <null>             |
+------+---------------------+-------------+--------------------+
```
</details>

### CODE REVIEW

File: aws/table_aws_secrets_manager_secret.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ARN").Transform(arnToAkas),
+			},
```
Comment (Paulami30) [inline aws/table_aws_secrets_manager_secret.go:142]: arrange the columns as title,tags,akas

File: aws/table_aws_secrets_manager_secret.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+	// Get call
+	op, err := svc.DescribeSecret(params)
+	if err != nil {
+		plugin.Logger(ctx).Debug("describeSecretsManagerSecret", "ERROR", err)
+		return nil, err
+	}
+	return op, nil
+}
+
+//// TRANSFORM FUNCTIONS
```
Comment (Paulami30) [inline aws/table_aws_secrets_manager_secret.go:217]:
```suggestion
//// TRANSFORM FUNCTION
```

File: docs/tables/aws_secrets_manager_secret.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+```sql
+select
+  name,
+  created_date,
+  description,
+  rotation_enabled
+from
+  aws_secrets_manager_secret
+where
+  rotation_enabled = 'false';
```
Comment (Paulami30) [inline docs/tables/aws_secrets_manager_secret.md:31]:
```suggestion
  not rotation_enabled;
```
rotation enabled is boolean

File: docs/tables/aws_secrets_manager_secret.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+
+### List secrets whose automatic rotation interval is greater tha 7 days
```
Comment (Paulami30) [inline docs/tables/aws_secrets_manager_secret.md:35]:
```suggestion
### List secrets whose automatic rotation interval is more than 7 days
```

File: aws/table_aws_secrets_manager_secret.go [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsSecretsManagerSecret(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_secrets_manager_secret",
```
Comment (Paulami30) [inline aws/table_aws_secrets_manager_secret.go:17]: the terraform resource name is `aws_secretsmanager_secret`

File: aws/table_aws_secrets_manager_secret.go [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "tags_src",
+				Description: "The list of user-defined tags associated with the secret.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Tags"),
+			},
+
+			/// Standard columns for all tables
```
Comment (Paulami30) [inline aws/table_aws_secrets_manager_secret.go:123]:
```suggestion
			// Standard columns for all tables
```

**Comment by anisadas** [comment]
<details><summary>Logs</summary>
<p>

```
SETUP: tests/aws_secretsmanager_secret []

PRETEST: tests/aws_secretsmanager_secret

TEST: tests/aws_secretsmanager_secret
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_secretsmanager_secret.named_test_resource: Creating...
aws_secretsmanager_secret.named_test_resource: Creation complete after 4s [id=arn:aws:secretsmanager:us-east-1:986325076436:secret:turbottest8741-hFxx9U]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Warning: Deprecated Attribute

  on variables.tf line 49, in resource "aws_secretsmanager_secret" "named_test_resource":
  49: resource "aws_secretsmanager_secret" "named_test_resource" {

Use the aws_secretsmanager_secret_rotation resource instead


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:secretsmanager:us-east-1:986325076436:secret:turbottest8741-hFxx9U
resource_name = turbottest8741

Running SQL query: test-get-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:secretsmanager:us-east-1:986325076436:secret:turbottest8741-hFxx9U"
    ],
    "arn": "arn:aws:secretsmanager:us-east-1:986325076436:secret:turbottest8741-hFxx9U",
    "deleted_date": null,
    "description": "Test secret.",
    "kms_key_id": null,
    "last_accessed_date": null,
    "last_rotated_date": null,
    "name": "turbottest8741",
    "owning_service": null,
    "partition": "aws",
    "primary_region": null,
    "region": "us-east-1",
    "replication_status": null,
    "rotation_enabled": null,
    "rotation_lambda_arn": null,
    "rotation_rules": null,
    "secret_versions_to_stages": null,
    "tags": {
      "foo": "bar"
    },
    "tags_src": [
      {
        "Key": "foo",
        "Value": "bar"
      }
    ],
    "title": "turbottest8741"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:secretsmanager:us-east-1:986325076436:secret:turbottest8741-hFxx9U",
    "name": "turbottest8741",
    "region": "us-east-1"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:secretsmanager:us-east-1:986325076436:secret:turbottest8741-hFxx9U"
    ],
    "arn": "arn:aws:secretsmanager:us-east-1:986325076436:secret:turbottest8741-hFxx9U",
    "name": "turbottest8741",
    "region": "us-east-1",
    "tags": {
      "foo": "bar"
    },
    "title": "turbottest8741"
  }
]
✔ PASSED

POSTTEST: tests/aws_secretsmanager_secret

TEARDOWN: tests/aws_secretsmanager_secret

SUMMARY:

1/1 passed.
```
</p>
</details>

**Review by Paulami30 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Comment by cbruno10** [comment]
@sayan133 I think this table should be called `aws_secretsmanager_secret`, as this is the name that Terraform uses (https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/secretsmanager_secret).

**Review by cbruno10 — APPROVED** [review]
Please rename this table (and all references

**Pull request merged**